ethcore/machine/src/test_helpers.rs

```diff
@@ -16,14 +16,16 @@
 
 //! Provide facilities to create `Machine` instances for testing various networks.
 
+use std::convert::TryFrom;
 use common_types::engines::params::CommonParams;
+use ethcore_builtin::Builtin;
 use ethjson;
 use crate::Machine;
 
 pub fn load_machine(reader: &[u8]) -> Machine {
 	let spec = ethjson::spec::Spec::load(reader).expect("chain spec is invalid");
 
-	let builtins = spec.accounts.builtins().into_iter().map(|p| (p.0.into(), From::from(p.1))).collect();
+	let builtins = spec.accounts.builtins().into_iter().map(|p| (p.0.into(), Builtin::try_from(p.1).expect("chain spec is invalid"))).collect();
```

Comment:
It would be cool if we could display all errors in the `builtin chain` spec if there were more than one error.

Currently, the error output looks like:
```bash
Thread 'main' panicked at 'Chain spec ethereum/foundation is invalid.: Msg("invalid builtin name: foo")'
```

Reply:
The `.into_iter().map(...).collect()` if the return type is `Result<_, _>` will also short circuit the map whenever there is an error. I'm not sure of a good way around that other than to create a new error variant that holds a vector of errors.

I tried to keep the same behavior as before as we will panic on the expect where it used to inside From::from.

Reply:
right, let's keep it out from this PR then